### ISSUE TITLE
Autocompletion should be aware of current queryText

### ISSUE THREAD

**Issue by huv1k** [graphql/graphiql#895]
I think autocompletion should be aware of current queryString, because you then get hit for fields what are currently in query. 

Here: https://github.com/graphql/graphql-language-service/blob/master/packages/interface/src/getAutocompleteSuggestions.js#L86

we get suggestion for everything available in schema and query string is completely ignored. https://i.imgur.com/heB9sRS.gif

**Comment by Neitsch** [comment]
@huv1k - feel free to submit a PR. However, please keep in mind that a user might want to query a field multiple times with different arguments.

**Comment by benjie** [comment]
I think we could see if they're using a field alias, and if not then we could omit it... or at least put it at the bottom of the list?

**Comment by acao** [comment]
yeah I appreciate seeing this proposal, maybe could eventually be an LSP configuration setting to that's disabled by default.

the reason being is that graphiql should not make assumptions about the implementation. from what i understand, any particular implementation could enforce multiple of the same field as valid or not, so it's not really something graphiql should be making decisions about, and we should default to providing the more flexible option which is allowing the same field multiple times.

**Comment by benjie** [comment]
Yeah; I agree with this analysis - it's perfectly valid to have the same field multiple times in a GraphQL query. A plugin would be the right way to deal with this. I'm going to rewrite the issue title.

**Comment by acao** [comment]
so, this shouldn’t be a plugin but a core LSP setting. it will be disabled by default as the graphql spec allows multiple fields so this is expected behavior in terms of the spec. free to open a PR for graphql language service interface! adding this option may be easier than you think!
